src/engraving/libmscore/keysig.cpp

```diff
@@ -315,6 +315,18 @@ void KeySig::draw(mu::draw::Painter* painter) const
     painter->setPen(curColor());
     for (const KeySym& ks: _sig.keySymbols()) {
         drawSymbol(ks.sym, painter, PointF(ks.pos.x(), ks.pos.y()));
+        // draw ledgerLines
+        double _spatium = spatium();
+        double xshift = (ks.sym == SymId::accidentalSharp) ? (_spatium * .15) : (_spatium * .25);
+        int _line = int(ks.pos.y() / _spatium);
+        while (_line < 0) {
+            drawSymbol(SymId::legerLine, painter, PointF((ks.pos.x() - xshift), (_line * _spatium)));
+            ++_line;
+        }
+        while (_line >= 5) {
```

Comment:
Here you assume that a staff has 5 lines, but that might not always be the case. Better use `staff() ? staff()->staffTypeForElement(this)->lines() : 5` instead of `5`. As an optimization, you could take that out of the for-loop and store it in a variable.

Reply:
Seems like you'll also need to account for staff line distance. Look at this failing VTest for example:
<img width="1440" alt="Schermafbeelding 2022-02-08 om 11 06 49" src="https://user-images.githubusercontent.com/48658420/152965433-e953487d-90d0-4320-8a01-6e047f6d48ed.png">
It's just a 5-line staff, but with a staff line spacing of 2.50 sp.

Reply:
Wouldn't `staff() ? staff()->staffTypeForElement(this)->lineDistance().val() : 1.0;` help here?

Reply:
Personally I would expect the distance between the ledger lines to be the same as the distance between the staff lines. That of course relates to the other bug you just discovered. If you would like to solve that bug yourself (while on it), I could assign it to you.

Reply:
Also, that VTest still seems to fail the same way as before... you will indeed need to use `staff() ? staff()->staffTypeForElement(this)->lineDistance().val() : 1.0;`, as Jojo suggests.

Reply:
Yes, distance should be same, as distance between lines, but as you can see, ledger lines corresponds with accidentals. I will chek something. Thank You.
One question `staff()->staffTypeForElement(this)->lineDistance().val()` or `staff()->lineDistance(tick())` would be OK?

Reply:
The latter looks simpler, but for symmetry alone I'd rather use the former?

Reply:
The former contains a small optimization compared to the latter, because it avoids a call to `tick()` in case there is only one staff type on that staff (which is the case in most scores, I'd think). For some elements, `tick()` is a very trivial call, but for others, it involves a bit more.